ACPs/189-seamless-l1-creation/README.md

```diff
@@ -0,0 +1,129 @@
+type CreateL1Tx struct {
+    // Metadata, inputs and outputs
+    BaseTx `serialize:"true"`
+    
+    // Chain configurations for the L1 (can be multiple)
+    Chains []ChainConfig `serialize:"true" json:"chains"`
+    
+    // Chain where the L1 validator manager lives
+    ManagerChainID ids.ID `serialize:"true" json:"managerChainID"`
+    
+    // Address of the L1 validator manager
+    ManagerAddress types.JSONByteSlice `serialize:"true" json:"managerAddress"`
+    
+    // Initial pay-as-you-go validators for the L1
+    Validators []*L1Validator `serialize:"true" json:"validators"`
+}
+```
+
+The `L1Validator` structure follows the same definition as in ACP-77.
+
+### **Transaction Processing**
+
+When a `CreateL1Tx` transaction is processed, the P-Chain performs the following operations atomically:
+
+1. Create a new L1
+2. Create chain records for each chain configuration in the `Chains` array.  
+3. Set up the L1 validator manager with the specified `ManagerChainID` and `ManagerAddress`  
+4. Register the initial validators specified in the `Validators` array
+
+### IDs
+
+* `subnetID`: The `subnetID` of the L1 is the transaction hash.
+* `blockchainID`: the `blockchainID` for each blockchain is is defined as the SHA256 hash of the 37 bytes resulting from concatenating the 32 byte subnetID with the `0x00` byte and the 4 byte chainIndex (index in the `Chains` array within the transaction)
+* `validationID`: The `validationID` for the initial validators added through CreateL1Tx is defined as the SHA256 hash of the 37 bytes resulting from concatenating the 32 byte subnetID with the `0x01` byte and 4 byte validatorIndex (index in the `Validators` array within the transaction).
+
+### **Restrictions and Validation**
+
+The `CreateL1Tx` transaction has the following restrictions and validation criteria:
+
+1. The `Chains` array must contain at least one chain configuration  
+2. The `ManagerChainID` must be a valid blockchain ID  
+3. The `ManagerAddress` must be a valid address for the specified chain type  
+4. Validator nodes must have unique NodeIDs within the transaction  
+5. Each validator must have a non-zero weight and a non-zero balance  
+6. The transaction inputs must provide sufficient AVAX to cover the transaction fee and all validator balances
+
+### **Warp Messages**
+
+After the transaction is accepted, the P-Chain must be willing to sign a `SubnetToL1ConversionMessage` with a `conversionID` corresponding to the new L1, similar to what would happen after a `ConvertSubnetToL1Tx`. This ensures compatibility with existing systems that expect this message, such as the validator manager contracts.
+
+## **Backwards Compatibility**
+
+This ACP introduces a new transaction type and does not modify the behavior of existing transaction types. Existing Subnets and L1s created through the three-step process will continue to function as before. This change is purely additive and does not require any changes to existing L1s or Subnets.
```

Comment:
I think it may be worth noting the three existing transaction types could be removed in a future ACP, but are left for now to ensure any systems currently using them have sufficient time to update to the new transaction type.